### PR TITLE
Bump deps to version 1.14

### PR DESCRIPTION
**Pull request by rcarrillocruz**

### CODE REVIEW

**Comment by squeed** [comment]
That was fast :-)

**Comment by squeed** [comment]
While you're at it, want to bump operator-sdk to 0.8? Annoyingly, you will have to figure out any changes to main.go from the operator-sdk templating.

**Comment by rcarrillocruz** [comment]
> While you're at it, want to bump operator-sdk to 0.8? Annoyingly, you will have to figure out any changes to main.go from the operator-sdk templating.

Yah, can look at it.

File: Gopkg.toml [hunk 1]
```diff
@@ -6,42 +6,41 @@ required = [
...
+  revision = "6ee68ca5fd8355d024d02f9db0b3b667e8357a0f"
 
 [[override]]
   name = "github.com/coreos/prometheus-operator"
   version = "=v0.26.0"
 
 [[override]]
   name = "sigs.k8s.io/controller-runtime"
-  version = "=v0.1.8"
+  version = "=v0.2.0-beta.1"
```
Comment (squeed) [inline Gopkg.toml]: don't bump this yet, this version is definitely not ready to go.

Comment (rcarrillocruz) [reply Gopkg.toml]: So yeah, this tripped me up.
The issue is that 0.2.0-beta.1 is the tag for controller-runtime to bump to Kubernetes 1.14.
Without pinning to that, we hit this on compile time:

```
# github.com/openshift/cluster-network-operator/vendor/sigs.k8s.io/controller-runtime/pkg/leaderelection
vendor/sigs.k8s.io/controller-runtime/pkg/leaderelection/leader_election.go:83:25: not enough arguments in call to resourcelock.New
        have (string, string, string, "github.com/openshift/cluster-network-operator/vendor/k8s.io/client-go/kubernetes/typed/core/v1".CoreV1Interface, resourcelock.ResourceLockConfig)
        want (string, string, string, "github.com/openshift/cluster-network-operator/vendor/k8s.io/client-go/kubernetes/typed/core/v1".CoreV1Interface, "github.com/openshift/cluster-network-operator/vendor/k8s.io/client-go/kubernetes/typed/coordination/v1".CoordinationV1Interface, resourcelock.ResourceLockConfig)
```

Check pkg/leaderelection/leader_election.go.
https://github.com/kubernetes-sigs/controller-runtime/commit/71cf6f6cf312e9111fc0ff1b411e19b91726359d#diff-2590281733c8f856c7a8fb319d5d3cc5R87

So I wonder if we should wait for controller-runtime to release a version that is non-beta, then we change the pin.
Reverting to 0.1.8 is a no-go.

Comment (squeed) [reply Gopkg.toml]: Ugh. @LiliC any suggestions?

Comment (LiliC) [reply Gopkg.toml]: Yes, we have a PR out in operator-sdk to switch to controller-runtime `0.2.0-x`, we haven't merged it yet as they are still in beta (from what I spoke with the maintainer they should be cutting the stable release soon if no bugs are found in testing). There are a lot of changes there, you can see in https://github.com/operator-framework/operator-sdk/pull/1388 PR if that is of any help. Sadly they did not release an intermediate version with bumping k8s to `1.14`.

I would agree to the following as we have things ready:

> So I wonder if we should wait for controller-runtime to release a version that is non-beta, then we change the pin.

**Comment by LiliC** [comment]
hi @rcarrillocruz so we merged the bump to k8s 1.14 into a branch on operator-sdk. You can see the PR https://github.com/operator-framework/operator-sdk/pull/1512. Maybe start testing against that commit from https://github.com/operator-framework/operator-sdk/tree/refactor/controller-runtime-v0.2.0, in this PR. We are blocked on controller-runtime cutting the final release of v0.2.0. But things should not change too much when we do merge that branch into operator-sdk master.

**Comment by rcarrillocruz** [comment]
oki, thx for the update @LiliC

**Pull request closed without merging**